### PR TITLE
Changed relationship

### PR DESCRIPTION
**Pull request by SkaeX**
#50.
- Changed User - School relationship from one-to-many to many-to-many.
- Also made necessary changes to the repository/test.

### CODE REVIEW

File: app/Repositories/Database/EloquentUserRepository.php [hunk 1]
```diff
@@ -45,7 +46,7 @@ public function getUsers()
      */
     public function getUsersBySchool($schoolId)
     {
-        return $this->findBy(['school_id' => $schoolId]);
+        return $this->transform(School::findOrFail($schoolId)->users);
```
Comment (BrainMaestro) [inline app/Repositories/Database/EloquentUserRepository.php]: This needs to be paginated.

File: tests/unit/Repositories/Database/EloquentUserRepositoryTest.php [hunk 2]
```diff
@@ -40,14 +40,15 @@ public function testGetUsers()
     public function testGetUsersBySchool()
     {
         $school = factory(Fce\Models\School::class)->create();
-        $users = factory(Fce\Models\User::class, 2)->create([
-            'school_id' => $school->id,
-        ]);
+        $users = factory(Fce\Models\User::class, 2)->create()->each(function ($u) use ($school) {
```
Comment (BrainMaestro) [inline tests/unit/Repositories/Database/EloquentUserRepositoryTest.php]: `$user` is clearer than `$u` right?

**Comment by SkaeX** [comment]
Approved.

File: app/Repositories/Database/EloquentUserRepository.php [hunk 3]
```diff
@@ -45,7 +46,7 @@ public function getUsers()
      */
     public function getUsersBySchool($schoolId)
     {
-        return $this->findBy(['school_id' => $schoolId]);
+        return $this->transform(School::findOrFail($schoolId)->users()->paginate(15, ['*'], 'page', 1));
```
Comment (BrainMaestro) [inline app/Repositories/Database/EloquentUserRepository.php]: We might eventually need to change this.

**Comment by BrainMaestro** [comment]
Approved

File: app/Repositories/Database/EloquentUserRepository.php (hunk 3 above)
Comment (SkaeX) [inline app/Repositories/Database/EloquentUserRepository.php]: I agree.

**Pull request merged**